### ISSUE TITLE
Output a history xml parameter file

### ISSUE THREAD

**Issue by kovenock** [EDmodel/ED2#202]
Hello,

I am hoping to modify pft parameters using an xml parameter file. In a previous thread (issue #173) @mdietze explained that doing a test run will generate a history xml file in your out directory, which can then be copied and modified to make parameter changes. I have completed a test run but do not have an .xml file in my run directory.

Is there a setting I need to update to output a history xml file with pft parameters used in a run? Or, a place to specify a pathway for the xml file?

Thanks!

**Comment by mdietze** [comment]
The generated history xml should be in your output directory, not your run directory.

It doesn't matter where you put the xml file, but you do need to specify it's location in the ED2IN

**Comment by kovenock** [comment]
@mdietze thanks for your reply! 

I have not found a history xml file in the output directories NL%FFILOUT and NL%SFILOUT specified in ED2IN. I've also searched within my higher level directories and not found it. Is there another output directory I should look for? 

Are there other ED2IN setting that are essential to outputting a history xml file? For NL%IEDCNFGF, I've tried leaving it empty (NL%IEDCNFGF=''), specifying a filename, specifying a path with a filename, and specifying a path without a filename. None of these settings produced a history xml.

Alternatively, is there a different way to output parameter values so that I can write my own history xml from them? I'm open to any suggestions of what to try next.

Thanks again for your help and insight!

**Comment by mdietze** [comment]
Not sure why it's not generating the file for you -- shows up as history.xml in our folders (example attached).  IEDCNFG is the input path for reading the xml -- I don't think it needs to be set to cause the file to write. Worst case you can find all the default parameter files in ed_params.f90
[history.xml.txt](https://github.com/EDmodel/ED2/files/1174535/history.xml.txt)

**Comment by kovenock** [comment]
Thanks again, @mdietze!

In the end I was able to generate a history xml file for my run using the following method:

I started a simulation with a very simple xml file (just one pft parameter change). This run generated a history xml file with all parameters for the run. I ran one other simulation that changed a single parameter for a different pft. I then combined information from the history xml files from these two runs (parameters for the unchanged pfts from each run) to create a full default history xml file for my initial run. Hope this helps if anyone else has trouble generating a history xml!
